docs/latest/design/request-authentication.md

```diff
@@ -0,0 +1,537 @@
+
+The following JWT Authentication filter `providers` configuration is created from the above Authentication.
+
+```yaml
+providers:
+   example:
+     issuer: https://www.example.com
+     audiences:
+     - foo.com
+     remote_jwks:
+       http_uri:
+         uri: https://foo.com/jwt/public-key/jwks.json
+         cluster: example_jwks_cluster
+         timeout: 1s
+```
+
+The following JWT Authentication filter `rules` configuration is created from the above HTTPRoute.
+
+```yaml
+rules:
+  - match:
+      prefix: /foo
+    requires:
+      provider_name: default-example-example
+```
+
+### Example 2: Two HTTPRoutes with Different Authentications
+
+The following example contains:
+* Two HTTPRoutes with different hostnames.
+* Each HTTPRoute references a different Authentication
+* Each Authentication contains a different JWT provider.
+
+```yaml
+apiVersion: security.gateway.envoyproxy.io/v1alpha1
+kind: Authentication
+metadata:
+  name: example1
+spec:
+  type: JWT
+  jwtProviders:
+  - name: example1
+    issuer: https://www.example1.com
+    audiences:
+    - foo.com
+    remoteJwks:
+      uri: https://foo.com/jwt/public-key/jwks.json
+---
+apiVersion: security.gateway.envoyproxy.io/v1alpha1
+kind: Authentication
+metadata:
+  name: example2
+spec:
+  type: JWT
+  jwtProviders:
+    - name: example2
+      issuer: https://www.example2.com
+      audiences:
+        - bar.com
+      remoteJwks:
+        uri: https://bar.com/jwt/public-key/jwks.json
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: example1
+spec:
+  hostnames:
+    - www.example1.com
+  parentRefs:
+    - group: gateway.networking.k8s.io
+      kind: Gateway
+      name: eg
+  rules:
+    - matches:
+        - path:
+            type: PathPrefix
+            value: /foo
+      filters:
+        - type: ExtensionRef
+          extensionRef:
+            group: security.gateway.envoyproxy.io
+            kind: Authentication
+            name: example1
+      backendRefs:
+        - name: backend
+          port: 3000
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: example2
+spec:
+  hostnames:
+    - www.example2.com
+  parentRefs:
+    - group: gateway.networking.k8s.io
+      kind: Gateway
+      name: eg
+  rules:
+    - matches:
+        - path:
+            type: PathPrefix
+            value: /bar
+      filters:
+        - type: ExtensionRef
+          extensionRef:
+            group: security.gateway.envoyproxy.io
+            kind: Authentication
+            name: example2
+      backendRefs:
+        - name: backend2
+          port: 3000
+```
+
+The following xDS configuration is created from the above example resources:
+
+```yaml
+configs:
+...
+dynamic_listeners:
+  - name: default-eg-http
+    ...
+        default_filter_chain:
+          filters:
+          - name: envoy.filters.network.http_connection_manager_1
```

Comment:
This won't work, HCM is a terminal filter, so any filter after that won't be effect and be rejected.

Reply:
@lizan I have updated the JWT xDS configuration example, PTAL.